src/main/java/teammates/ui/controller/AdminAccountManagementPageAction.java

```diff
@@ -15,39 +15,40 @@
 
     @Override
     protected ActionResult execute() throws EntityDoesNotExistException {
-        
         new GateKeeper().verifyAdminPrivileges(account);
         
-        @SuppressWarnings("deprecation") //This method is deprecated to prevent unintended usage. This is an intended usage.
-        List<InstructorAttributes> allInstructorsList = logic.getAllInstructors();
-        @SuppressWarnings("deprecation") //This method is deprecated to prevent unintended usage. This is an intended usage.
-        List<AccountAttributes> allInstructorAccountsList = logic.getInstructorAccounts();
-        
-        boolean isToShowAll = this.getRequestParamAsBoolean("all");
+        String instructorGoogleId = this.getRequestParamValue("googleId");
         
         Map<String, ArrayList<InstructorAttributes>> instructorCoursesTable = new HashMap<String, ArrayList<InstructorAttributes>>();
         Map<String, AccountAttributes> instructorAccountsTable = new HashMap<String, AccountAttributes>();
         
-        for(AccountAttributes acc : allInstructorAccountsList){
-            instructorAccountsTable.put(acc.googleId, acc);
+        if (instructorGoogleId == null) {
+            instructorGoogleId = "";
         }
+        List<InstructorAttributes> instructorsList = logic.getInstructorsForGoogleId(instructorGoogleId);
+        AccountAttributes instructorAccount = logic.getAccount(instructorGoogleId);
         
-        for(InstructorAttributes instructor : allInstructorsList){
-            ArrayList<InstructorAttributes> courseList = instructorCoursesTable.get(instructor.googleId);
-            if (courseList == null){
-                courseList = new ArrayList<InstructorAttributes>();
-                instructorCoursesTable.put(instructor.googleId, courseList);
-            }
+        boolean isToShowAll = this.getRequestParamAsBoolean("all");
+        boolean isAccountExisting = instructorAccount != null && !instructorsList.isEmpty();
```

Comment:
@bamboo3250, Hmm, I wonder if the check that <code>!instructorsList.isEmpty();</code> is really needed? What will happen if the account exist in the database, but the account does not have any instructors with it?